src/diffusers/models/attention.py

```diff
@@ -434,6 +434,7 @@ def __init__(
         cross_attention_dim: Optional[int] = None,
     ):
         super().__init__()
+        self.time_mix_inner_dim = time_mix_inner_dim
```

Comment:
@DN6 I think it would really help if every module could store its init parameters as attributes. It helps with customizing models and I've experienced roadblocks in getting the correct dimensions when modding different layers for experimentation. Here's an example to demonstrate the use case:

```py
for _, module in self.named_modules():
    if isinstance(module, ResnetBlock2d):
        new_layer = nn.Linear(module.in_channels, module.out_channels)
        module.add_module("new_layer", new_layer)
  
        new_forward = custom_resnetblock2d_forward.__get__(module, module.__class__)
        setattr(module, "forward", new_forward)
```

Many modelling blocks already do this, as is the case with ResnetBlock2D, but many don't such as TemporalBasicTransformerBlock. It would help if it was consistent across all modelling components. WDYT?